legal-api/src/legal_api/models/party_role.py

```diff
@@ -128,18 +128,19 @@ def get_active_directors(business_id: int, end_date: datetime) -> list:
     @staticmethod
     def get_party_roles(business_id: int, end_date: datetime = None, role: str = None) -> list:
         """Return the parties that match the filter conditions."""
-        # TODO: remove filter that excludes officer when we have plans to deal with it
         party_roles = db.session.query(PartyRole). \
-            filter(PartyRole.business_id == business_id). \
-            filter(PartyRole.role != PartyRole.RoleTypes.OFFICER.value)
+            filter(PartyRole.business_id == business_id)
+
+        # TODO: rework the Officers section; change-on-change to still bring back officers
+        if not role or role != PartyRole.RoleTypes.OFFICER.value:
+            party_roles = party_roles.filter(PartyRole.role != PartyRole.RoleTypes.OFFICER.value)
+        if role is not None:
+            party_roles = party_roles.filter(PartyRole.role == role.lower())
```

Comment:
Think there's an issue with this logic.  If you pass in `role` = "director", it will return all non officer roles?

Reply:
yes, that's what it used to do.
I just changed it, from the last change that totally removed the ability to get officers.